File: core/src/main/java/com/ctrip/ferriswheel/core/action/RemoveColumns.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.core.intf.Action;

public class RemoveColumns extends TableAction implements Action {
    private int columnIndex;
    private int nColumns;

    public RemoveColumns() {
    }

    public RemoveColumns(String sheetName, String tableName, int columnIndex, int nColumns) {
        super(sheetName, tableName);
        this.columnIndex = columnIndex;
        this.nColumns = nColumns;
    }

    @Override
    public String toString() {
        return "RemoveColumns{" +
                "sheetName='" + getSheetName() + '\'' +
                ", tableName=" + getTableName() +
                ", columnIndex=" + columnIndex +
                ", nColumns=" + nColumns +
                '}';
    }

    public int getColumnIndex() {
        return columnIndex;
    }

    public void setColumnIndex(int columnIndex) {
        this.columnIndex = columnIndex;
    }

    public int getnColumns() {
        return nColumns;
    }

    public void setnColumns(int nColumns) {
        this.nColumns = nColumns;
    }
}

File: ui/react/src/action/RemoveColumns.ts
import TableAction from "./TableAction";
import Action from "./Action";

class RemoveColumns extends TableAction {
    public columnIndex: number;
    public nColumns: number;

    public static deserialize(input: any) {
        const sheetName: string = input.sheetName;
        const tableName: string = input.tableName;
        const columnIndex: number = input.columnIndex;
        const nColumns: number = input.nColumns;
        return new RemoveColumns(
            sheetName,
            tableName,
            columnIndex,
            nColumns
        );
    }

    constructor(
        sheetName: string,
        tableName: string,
        columnIndex: number,
        nColumns: number) {

        super(sheetName, tableName);
        this.columnIndex = columnIndex;
        this.nColumns = nColumns;
    }

    public wrapper(): Action {
        const action = new Action();
        action.removeColumns = this;
        return action;
    }

}

export default RemoveColumns;
